processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -707,6 +719,33 @@ protected void putToRow(ByteBuffer key, ResultRow resultRow)
         );
       }
     }
+
+    /**
+     * check if a collection of columns will operate correctly with {@link LimitedBufferHashGrouper} for query limit
+     * pushdown
+     */
+    private boolean canPushdownLimit(Stream<String> columnNames)
```

Comment:
IMO it's more elegant if this method doesn't exist, and the caller does the allMatch.